### PR TITLE
Merging solar system changes

### PR DESCRIPTION
**Pull request by rshingavi97**
I have merged an very initial level solar system changes.
Kindly review it.

### CODE REVIEW

File: common/include/Renderer.h [hunk 1]
```diff
@@ -43,7 +43,7 @@ namespace Interfaces
         SCENE_DIODE,
         SCENE_TRIANSISTOR,
         SCENE_GATES,
-        SCENE_CHIPS
+        SCENE_CHIPS,
```
Comment (mknilesh2016) [inline common/include/Renderer.h]: This comma should be removed.

File: src/features/RendererRegistrar.cpp [hunk 2]
```diff
@@ -19,17 +20,23 @@ namespace Features
 
         static SceneToRenderersMap renderers;
         INIT_RENDERER_MAP(renderers)
+			
+#ifdef _TEST
```
Comment (mknilesh2016) [inline src/features/RendererRegistrar.cpp]: Can you help correcting the formatting here?

File: src/features/RendererRegistrar.cpp [hunk 3]
```diff
@@ -19,17 +20,23 @@ namespace Features
...
-
-            BEGIN_SCENE_TYPE(SCENE_TYPE_TEST1)
-                REGISTER_RENDERER(TestRenderer2)
-                // other renderer?
-            END_SCENE_TYPE()
+			BEGIN_SCENE_TYPE(SCENE_TYPE_TEST1)
+				REGISTER_RENDERER(TestRenderer2)
+			// other renderer?
+			END_SCENE_TYPE()
+
```
Comment (mknilesh2016) [inline src/features/RendererRegistrar.cpp]: you may want to remove this trailing line.

File: src/features/SolarSystem/SolarRender.h [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+#include <gl\GLU.h>
+#include"features\SolarSystem\SolarResource.h"
+namespace Features
+{
+	using namespace Interfaces;
+
+	class SolarSystemRenderer : public IRenderer
+	{
+	private:
+		float		g_Angle;
```
Comment (mknilesh2016) [inline src/features/SolarSystem/SolarRender.h]: As a generic guideline, can we use GL types?

e.g. in this case, can this be changed to GLfloat?

File: src/features/SolarSystem/SolarRender.h [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+			g_Mercury = g_Mercury + 0.7f;
+			g_Venus = g_Venus + 0.3f;
+			g_Earth = g_Earth + 0.5f;
+			g_Mars = g_Mars + 0.3f;
+			g_Jupiter = g_Jupiter + 0.5f;
+			g_Saturn = g_Saturn + 0.6f;
+			g_Uranus = g_Uranus + 0.7f;
+			g_Neptune = g_Neptune + 0.5f;
+		}
+		int LoadTexture(GLuint *texture, TCHAR resource[])
```
Comment (mknilesh2016) [inline src/features/SolarSystem/SolarRender.h]: Wouldn't be better to use SOIL for loading the image file?
If you use SOIL, then you can load more compressed image types which can save disk-IO and get us faster loading time.

File: src/features/SolarSystem/SolarRender.h [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+			LoadTexture(&gbTexture[4], MAKEINTRESOURCE(JUPITER_BITMAP));
+			LoadTexture(&gbTexture[5], MAKEINTRESOURCE(SATURN_BITMAP));
+			LoadTexture(&gbTexture[6], MAKEINTRESOURCE(URANUS_BITMAP));
+			LoadTexture(&gbTexture[7], MAKEINTRESOURCE(NEPTUNE_BITMAP));
+			LoadTexture(&gbTexture[8], MAKEINTRESOURCE(SUN_BITMAP));
+			LoadTexture(&gbTexture[9], MAKEINTRESOURCE(STARS_BITMAP));
+
+
+			for (int i = 0; i < 9; i++)
+				pQuadric[i] = gluNewQuadric();
```
Comment (mknilesh2016) [inline src/features/SolarSystem/SolarRender.h]: shouldn't we check for the return type of the data returned by gluNewQuadric for the initializations? Fail the initialization if any of them are invalid.

File: src/features/SolarSystem/SolarRender.h [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+		float 		g_Jupiter;
+		float 		g_Saturn;
+		float 		g_Uranus;
+		float 		g_Neptune;
+		GLuint		gbTexture[10];
+		GLUquadric	*pQuadric[9];
+	public:
+		void update()
+		{
+			g_Angle = g_Angle + 0.1f;
```
Comment (mknilesh2016) [inline src/features/SolarSystem/SolarRender.h]: Can we move the implementation into its dedicated file before merging to master?

File: SolarRender.h [hunk 8]
```diff
@@ -0,0 +1,354 @@
+#pragma once
+
```
Comment (mknilesh2016) [inline SolarRender.h]: This appears to be duplicate file and should be removed before merging to master.

**Review by mknilesh2016 — CHANGES_REQUESTED** [review]
Thanks for the initial implementations of the solar system. This gives us a head start.

Most of my comments are process' and coding oriented to it is likely possible that you may have different views on the same.

Feel free to revert back if you require specific explanation of any of these comments.

**Comment by rshingavi97** [comment]
Hello Nilesh,

I just saw your comments.
I will work on it.





Regards,
Rupesh Sukhalal Shingavi,

M:9823340767

“Do help and get help”

File: common/include/Renderer.h (hunk 1 above)
Comment (rshingavi97) [reply common/include/Renderer.h]: I removed

File: SolarRender.h (hunk 8 above)
Comment (rshingavi97) [reply SolarRender.h]: I removed it.

File: src/features/RendererRegistrar.cpp (hunk 2 above)
Comment (rshingavi97) [reply src/features/RendererRegistrar.cpp]: I did it based on my understanding. I might not be done according to your perspective. 
Thanks

File: src/features/RendererRegistrar.cpp (hunk 3 above)
Comment (rshingavi97) [reply src/features/RendererRegistrar.cpp]: I did it.

File: src/features/SolarSystem/SolarRender.h (hunk 4 above)
Comment (rshingavi97) [reply src/features/SolarSystem/SolarRender.h]: I accepted your suggestion, Changed it .

File: src/features/SolarSystem/SolarRender.h (hunk 7 above)
Comment (rshingavi97) [reply src/features/SolarSystem/SolarRender.h]: It is done.

File: src/features/SolarSystem/SolarRender.h (hunk 6 above)
Comment (rshingavi97) [reply src/features/SolarSystem/SolarRender.h]: It is done

**Comment by rshingavi97** [comment]
Today I have merged all the changes suggested by Nilesh.
Kindly merge the changes.

**Pull request closed without merging**